app/views/manufacturers/_form.html.slim

```diff
@@ -0,0 +1,20 @@
+= form_for(@manufacturer) do |f|
+  - if @manufacturer.errors.any?
+    #error_explanation
+      h2
+        = pluralize(@manufacturer.errors.count, "error")
+        |  prohibited this role from being saved:
+      ul
+        - @manufacturer.errors.full_messages.each do |message|
+          li
+            = message
+  .form-group
+    = f.label :name
+    br
+    = f.text_field :name, class: "form-control"
+  #models.form-group
+    = f.fields_for :models do |model|
+      = render 'model_fields', f: model
+    .links
+      = link_to_add_association 'Add Model', f, :models, class: "btn btn-primary"
+  = f.submit class: "btn btn-primary"
```

Comment:
Because this button is used to update the manufacturers (i.e. Add, Edit, etc.), I found it a bit difficult to use if there are many models located between the manufacturers' names and this button.

For example, if you try to add more than 10 models for a manufacturer, then you can see that the text field for manufacturer's name and this button is very far. It will be easier if this button is put below the manufacturer text field immediately.

Reply:
I updated it.